File: back/src/models/PhilopateerRequest.js
import mongoose from 'mongoose';

const philopateerRequestSchema = new mongoose.Schema(
  {
    requestType: { type: String, required: true, index: true }, // poster, video, montage, office, photography, sound
    requesterUsername: { type: String, required: true, index: true },
    requesterName: { type: String, required: true },
    requesterOsra: { type: String, default: '' },
    details: { type: String, default: '' },
    requiredDate: { type: String, required: true }, // YYYY-MM-DD
    files: { type: [String], default: [] }, // base64 strings or file paths
    photographyDetails: {
      whatToPhotograph: { type: String, default: '' },
      timings: { type: String, default: '' },
      hasFullEquipment: { type: Boolean, default: false }
    },
    status: { type: String, default: 'pending', index: true }, // pending, approved, rejected, in_progress
    seen: { type: Boolean, default: false, index: true }
  },
  {
    timestamps: true,
    collection: 'philopateer_requests'
  }
);

const PhilopateerRequest = mongoose.models.PhilopateerRequest || mongoose.model('PhilopateerRequest', philopateerRequestSchema);
export default PhilopateerRequest;

File: back/Models/PhilopateerRequest.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("philopateer_requests")]
    public class PhilopateerRequest
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = string.Empty;

        [Column("requestType")]
        public string RequestType { get; set; } = string.Empty;

        [Column("requesterUsername")]
        public string RequesterUsername { get; set; } = string.Empty;

        [Column("requesterName")]
        public string RequesterName { get; set; } = string.Empty;

        [Column("requesterOsra")]
        public string RequesterOsra { get; set; } = string.Empty;

        [Column("details")]
        public string Details { get; set; } = string.Empty;

        [Column("requiredDate")]
        public string RequiredDate { get; set; } = string.Empty;

        [Column("filesJson")]
        public string FilesJson { get; set; } = "[]"; // Serialized JSON string array of file names or paths

        [Column("photographyDetailsJson")]
        public string PhotographyDetailsJson { get; set; } = "{}"; // Serialized JSON of photography details object

        [Column("status")]
        public string Status { get; set; } = "pending";

        [Column("seen")]
        public bool Seen { get; set; } = false;

        [Column("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
